components/factory/forms/BurnForm.tsx

```
@@ -12,68 +1,99 @@
+    amount: Yup.number()
+      .positive('Amount must be positive')
+      .required('Amount is required')
+      .max(1e12, 'Amount is too large')
+      .test('max-balance', 'Amount exceeds balance', function (value) {
+        return value <= balanceNumber;
+      }),
+    recipient: Yup.string().required('Recipient address is required').manifestAddress(),
+  });
 
   const handleBurn = async () => {
     if (!amount || isNaN(Number(amount))) {
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)